src/components/TextInput.jsx

```diff
@@ -114,32 +114,21 @@ export default class TextInput extends Component {
 
     if (!this.props.allowNewLine && value.match(newLineRegex) !== null) {
       value = event.target.value.replace(newLineRegex, ' ');
-
-      // controlled textarea must have value
```

Comment:
Hi Nik,

Yep I did got that but after I had done these changes. U r right we need to take closer look at this.